### PR TITLE
Repro/dashboard searchbox permissions

### PR DESCRIPTION
**Pull request by dacort**
~~This is a test. This is only a test.~~

Reproduces #8472

@tlrobinson @paulrosenzweig Would love your feedback here. This PR adds a Cypress test case for a desired state, as opposed to a current state. This is intended to help us ensure that if we have a repro for an issue, we can commit that code ahead of time and use it to validate the fix. Places where I'm looking for feedback:

1) Directory structure - Right now I'm dropping these in `frontend/test/metabase/support`. That way we can tell if a specific test is related to a support/repro case, which will allow us to confidently disable it if starts failing for some reason.
2) Using `it.skip()` to bypass the test. My thought here is that until we fix the issue, the test will fail but we don't want that to fail our CI. This requires whomever fixes the issue to enable this test. 
3) `describe` comment - I'm currently adding the GitHub issue in the `describe` comment. Let me know if that format is cool.

### CODE REVIEW

**Comment by dacort** [comment]
I'm also not sure why `fe-linter-prettier` is failing. When I run `yarn prettier` locally, the only thing it changes is the first line from a single-line import to a multi-line import. Even though the rest of our Cypress tests use a single-line import... 🤔 

```diff
-import { signIn, restore, popover, selectDashboardFilter } from "__support__/cypress";
+import {
+  signIn,
+  restore,
+  popover,
+  selectDashboardFilter,
+} from "__support__/cypress";
```

**Comment by tlrobinson** [comment]
@dacort I love the idea of adding a reproducing cypress test as the first step in fixing an issue. 

Prettier breaks up long lines. In general if `fe-linter-prettier` is failing, just run `yarn prettier` and commit the results. Don't question `prettier` :) If this is annoying I suggest either setting up editor support to run prettier on save, or get in the habit of running `yarn prettier` before committing.

**Comment by paulrosenzweig** [comment]
This is a great idea!

My one thought would be that we might want to move the test once it's fixed in addition to enabling it. Organizing current bugs by issue ID is helpful, but it's probably nicer to have existing tests grouped by what part of the app they test. Keeping the issue ID as we move it, would let us still associate it later without looking at git history.

Given our sparse test coverage, I'd assume most of these will feel like something that should be a normal test of our functionality rather than something very specific to a single bug. But, if that's not the case then maybe we should leave them categorized by issue.

**Comment by dacort** [comment]
@paulrosenzweig Yea, I think that's reasonable re: moving it. Ideally the functionality encompassed should be maintained going forward. 

A couple followup questions:

- Is `master` where these should be merged into? That might make it a little annoying to have one PR to fix and another to remove the test file. 
- Any preference on letting folks know a repro exists? Just link the two issues via a comment? Add a tag?

File: frontend/test/metabase/support/dashboard_data_permissions.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  // We should get a suggested response and be able to click it if we're an admin
+  if (suggests) {
+    cy.contains("Category").type("Aero");
+    cy.contains("Aerodynamic").click();
+  } else {
+    cy.contains("Category").type("Aerodynamic Bronze Hat");
+  }
+  cy.contains("Add filter").click();
+  cy.contains("Rows 1-1 of 96");
```
Comment (nemanjaglumac) [inline frontend/test/metabase/support/dashboard_data_permissions.cy.spec.js:20]: I already ran into problems with a similar assertion. The number of **shown** rows ("1-1") directly depends on the size of the card. What we're interested in is the total number of rows ("96").

I'll update this to use Regex.

**Review by nemanjaglumac — APPROVED** [review]
@dacort is approving this through me :D

**Comment by nemanjaglumac** [comment]
Rebased and force-pushed on top of the current `master` because the test was still referencing the old icons and UI.

**Pull request merged**